[Desmond Tan Jing Shen] iP

mazx4960 opened the pull request:
### Jarvis
> *Your personal assistant.*

Jarvis frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/mazx4960/ip/releases/tag/v0.1).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

mazx4960 committed:
Updated for level 1

mazx4960 committed:
Updated for level 2

mazx4960 committed:
Updated for level 3

mazx4960 committed:
Updated for level 3

mazx4960 committed:
Updated for level 4

mazx4960 committed:
Updated for level 4

mazx4960 committed:
Updated for A-TextUiTesting

mazx4960 committed:
Updated for level 5

mazx4960 committed:
Updated for level 6

mazx4960 committed:
Updated for level 7

mazx4960 committed:
Updated for level 8

mazx4960 committed:
Updated for A-MoreOOP

mazx4960 committed:
Updated for A-Packages

mazx4960 committed:
Updated for A-Packages

mazx4960 committed:
Updated for A-JUnit

mazx4960 committed:
Updated for v0.1

mazx4960 committed:
Updated for A-JavaDoc

mazx4960 committed:
Update for A-CodingStandard

mazx4960 committed:
Updated for level 9

mazx4960 committed:
Add tests for find

mazx4960 committed:
Update README.md

mazx4960 committed:
Add gradle support

mazx4960 committed:
Add gradle support

mazx4960 committed:
Merge branch 'add-gradle-support'

mazx4960 committed:
Added support for checkstyle and fixed styling

mazx4960 committed:
Added support for checkstyle and fixed styling

mazx4960 committed:
Merge branch 'branch-A-CheckStyle'

chuashihong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @return boolean of whether the instance of running
+     */
+    public boolean isRunning() {
+        return this.status.equals(STATUS_RUNNING);
+    }
```
Great! Boolean method sounds like boolean

chuashihong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+    private void handler() {
+        ui.greet();
+
+        while (this.status.equals(STATUS_RUNNING) && ui.hasNextCmd()) {
+            try {
+                String cmdString = ui.nextCmd();
+                Command cmd = Parser.parse(cmdString);
+                cmd.execute(dukeList, ui);
+                if (cmd.isExit()) {
+                    this.status = STATUS_STOPPED;
+                }
+            } catch (DukeException e) {
+                ui.printMsg(e.getMessage());
+            }
+        }
```
No coding standard violation detected. Great Work!

chuashihong commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,164 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.exceptions.InvalidTaskException;
+import duke.tasks.Task;
```
A spacing between different import, great job!

chuashihong commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,164 @@
+import duke.exceptions.InvalidTaskException;
+import duke.tasks.Task;
+
+public class DukeList {
```
Perhaps you can explain what this class does

chuashihong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,141 @@
+package duke;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.QuitCommand;
+import duke.commands.UnmarkCommand;
```
If all the files in the commands is going to be imported, maybe you could just use
`import duke.commands.*;`
(but I am not sure whether this violates the coding standard but I feel it is more clean)

chuashihong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,141 @@
+    }
+
+    /**
+     * Checks if the string is numeric.
```
The javaDoc follows the correct grammar!

chuashihong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,141 @@
+    }
+
+    /**
+     * Get keyword from the args.
```
javaDoc grammar, Get-> Gets

chuashihong commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,188 @@
+        String taskType = taskArr[0];
+        String[] params;
+
+        switch (taskType) {
```
Switch case follows coding standard! Good Job!

chuashihong commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,188 @@
+        boolean taskCompleted = Objects.equals(taskArr[1], "x");
+        String taskDescription = taskArr[2];
+
+        switch (taskCode) {
```
Switch case statement follows the coding standard! Good Job!

chuashihong commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,188 @@
+     * @param keyword String to search for
+     * @return true if the description contains the keyword
+     */
+    public boolean matches(String keyword) {
```
Perhaps you can change the method name to
`has_contains_keyword(String keyword)`
so that it sounds more like boolean

chuashihong commented on src/test/java/duke/test/ParserTest.java:
```diff
@@ -0,0 +1,56 @@
+import org.junit.jupiter.api.Test;
+
+import duke.Parser;
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.QuitCommand;
+import duke.commands.UnmarkCommand;
```
Perhaps could just use 

`duke.command.*`

if all the package files in the commands are needed

chuashihong reviewed (commented):
Overall the coding standard are well followed!! Keep up the good work!!
The structure of the project is also very clean

mazx4960 committed:
Updated for level 10

mazx4960 committed:
Updated for level 10

mazx4960 committed:
Lint

mazx4960 committed:
Updated find command to use varargs

mazx4960 committed:
Added CI

mazx4960 committed:
Make gradlew executable

mazx4960 committed:
Make runtest executable

mazx4960 committed:
Fixed CI

mazx4960 committed:
Fixed CI

mazx4960 committed:
Fixed CI

mazx4960 committed:
Fixed CI

bloodofme commented on src/main/java/jarvis/utils/TextUI.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @return the next command
+     */
+    public String nextCmd() {
```
Perhaps it would improve the code quality if the method name was used to describe what the method does. For example, changing the name to `getNextCmd` 🤔

bloodofme commented on src/main/java/jarvis/utils/Storage.java:
```diff
@@ -0,0 +1,95 @@
+    private static final String FILENAME = "duke.txt";
+    private static final Path FILE_PATH = Paths.get(ROOT_FOLDER, DATA_DIR, FILENAME);
+
+    public Storage() {
```
Default constructors are automatically generated, would it be better to remove this line ?

bloodofme commented on src/main/java/jarvis/tasks/Task.java:
```diff
@@ -0,0 +1,188 @@
+     * @param keyword String to search for
+     * @return true if the description contains the keyword
+     */
+    public boolean matches(String keyword) {
```
Perhaps it would be better to think of a more descriptive method name

bloodofme reviewed (commented):
Overall, code was readable and excellent JavaDocs 👍🏼

mazx4960 committed:
Full commit message

Make commit messages more informative

A longer commit body is included

mazx4960 committed:
Add assertions to handle unexpected errors

of method in the Task class: added an assertion to check that the
taskArr is larger or equals to 1

The variable taskType extracts the first value from the taskArr, hence
taskArr needs to have at least one element

By checking that the length of the taskArr is at least 1, it would never
reach the state where taskType has no value to extract from taskArr

mazx4960 committed:
Refine code quality

Refactored various methods to make it more readable

Some methods might be too messy and difficult to read

mazx4960 committed:
Fixed checkstyle

mazx4960 committed:
Merge pull request #2 from mazx4960/branch-A-CodeQuality

Refine code quality

mazx4960 committed:
Merge pull request #1 from mazx4960/branch-A-Assertions

Add assertions to handle unexpected errors

mazx4960 committed:
Added tagging

mazx4960 committed:
Set theme jekyll-theme-cayman

mazx4960 committed:
Add user guide

mazx4960 committed:
Merge branch 'master' of https://github.com/mazx4960/ip

mazx4960 committed:
Update build.gradle file

mazx4960 committed:
Bug fixes
